### ISSUE TITLE
Warn for `#[allow()]`s which do not suppress anything

### ISSUE THREAD

**Issue by jgopel** [rust-lang/rust-clippy#16275]
### What it does

Imagine I have a function like this:

```rust
fn func(
    independent: f64,
    param1: i32,
    param2: &str,
    param3: u64,
    param4: bool,
    param5: i32,
    param6: f32,
) {}
```

In a setup that enables it, this would trigger `too_many_arguments`, so I might `#[allow(clippy::too_many_arguments)]` if I don't have a good solution to it at the time. At some point in the future, I may realize that I can write it like tihs:

```rust
#[allow(clippy::too_many_arguments)]
fn func(indepdent: f64, params: Params) {}
```

It's an easy mistake to forget to remove the `allow`, even in this case. In cases where the `allow` is less local it's even easier. This now leaves the code in a bit of a confusing state where there's an allow that doesn't do anything. For another developer who doesn't have the context of the walk of changes, it might cause them to pause and fruitlessly inspect the code.

I would propose that this allow should generate a clippy warning (which itself could be allowed, if desired). This ensures that code stays as clean as possible. A great beachhead would be just handling clippy `allow`s, but I think there's also value in handling, eg, `dead_code`s which have become incorrect.

Forgive me if there has been an issue created for this already - I searched for it and could not find one.

### Advantage

- Makes code cleaner
- Makes clippy even easier to use

### Drawbacks

- Potentially chatty in some cases where lints are suppressed even though they are not in place in exactly that moment, but may be in the future.

### Example

See above

### Comparison with existing lints

I'm not aware of anything similar in clippy currently.

### Additional Context

There is prior art for this in `pylint`'s `useless-suppression` lint.

**Comment by samueltardieu** [comment]
Isn't `#[expect(clippy::too_many_arguments)]` the thing you're looking for?

**Comment by jgopel** [comment]
This is exactly what I was looking for - thank you for pointing me in the right direction.
